navigator_helpers/data_models.py

```diff
@@ -299,28 +320,61 @@ def from_yaml(cls, yaml_str: str) -> "DataModel":
         data = yaml.safe_load(yaml_str)
         if "contextual_tags" in data:
             tags_data = data["contextual_tags"]["tags"]
-            data["contextual_tags"] = ContextualTags(tags=[ContextualTag(**tag) for tag in tags_data])
+            data["contextual_tags"] = ContextualTags(
+                tags=[ContextualTag(**tag) for tag in tags_data]
+            )
         if "data_source" in data:
             data["data_source"] = DataSource(**data["data_source"])
         return cls(**data)
 
     def to_yaml(self) -> str:
         """
-        Converts the DataModel object to a YAML string.
-
-        Returns:
-            str: YAML representation of the DataModel object.
+        Converts the DataModel object to a YAML string, ensuring fields are ordered as:
+        - generation_instructions
+        - fields
+        - contextual_tags (if present)
+        Followed by any additional fields.
         """
-        data = self.dict()
+        yaml_output = []
+
+        # Generation instructions
+        yaml_output.append("generation_instructions: |")
+        instructions_text = self.generation_instructions.replace("\n", "\n  ")
+        yaml_output.append(f"  {instructions_text}")
+
+        # Fields
+        yaml_output.append("fields:")
+        for field in self.fields:
+            yaml_output.append(f"  - name: {field.name}")
+            yaml_output.append(f"    type: {field.type}")
+            yaml_output.append(f"    description: |")
+            description_text = field.description.replace("\n", "\n      ")
+            yaml_output.append(f"      {description_text}")
```

Comment:
Indents may be tricky to track (especially if the value can have newlines like here), would it be possible here to create a dict representation of that YAML and then just do `yaml.dump`?